### ISSUE TITLE
FR: Move the id-token write permissions from the workflow level to the job level in the base workflows

### ISSUE THREAD

**Issue by MartinPetkov** [abcxyz/guardian#320]
### TL;DR

Move the `id-token: 'write'` permission in the [base workflows](https://github.com/abcxyz/guardian/tree/main/abc.templates/base-workflows/contents) from the workflow level to the job level.

### Detailed design

In the [base workflows](https://github.com/abcxyz/guardian/tree/main/abc.templates/base-workflows/contents), the following permission is defined at the workflow level:

```
permissions:
  id-token: 'write'
```

This is more of a precaution, but it would be better to move it to the job level, and only to jobs that need it.

IIUC the `id-token: 'write'` permissions is [only necessary for OIDC authentication to GCP](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-google-cloud-platform), so other jobs in the workflows don't need this ambient permission.

### Additional information

I tried this with someone and it failed with this error:

```
  Error: fatal: repository 'https://github.com/my-org/my-repo' not found
  Error: The process '/usr/bin/git' failed with exit code 128
```

So maybe it's not so straightforward of a change.
